prow/jira/jira.go

```diff
@@ -121,9 +134,27 @@ func (u userAgentSettingTransport) RoundTrip(r *http.Request) (*http.Response, e
 	return u.upstream.RoundTrip(r)
 }
 
+type clientUsedTransport struct {
+	used     bool
+	upstream http.RoundTripper
+}
+
+func (c clientUsedTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	c.used = true
+	return c.upstream.RoundTrip(r)
+}
+
 type client struct {
-	url      string
-	upstream *jira.Client
+	logger    *logrus.Entry
+	upstream  *jira.Client
+	transport *clientUsedTransport
```

Comment:
nit: can this be an interface - I don't think we necessarily want to expose the delegate or transporting function in this field

```go
type used interface {
    Used() bool
}
```

Reply:
Reasonable but segfaulting :) Thanks